dev/com.ibm.ws.messaging.comms.client/src/com/ibm/ws/sib/netty/jfapchannel/NettyIOConnectionContext.java

```diff
@@ -0,0 +1,131 @@
+	   }
+
+
+	   /** The connection reference */
+	   private NettyNetworkConnection conn = null;
+
+	   /**
+	    * @param tcpCtx
+	    */
+	   public NettyIOConnectionContext(NettyNetworkConnection conn)
+	   {
+	      if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) SibTr.entry(this, tc, "<init>", new Object[]{conn});
+	      this.conn = conn;
+	      if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) SibTr.exit(tc, "<init>");
+	   }
+
+	   /**
+	    *
+	    * @see com.ibm.ws.sib.jfapchannel.framework.IOConnectionContext#getLocalAddress()
+	    */
+	   public InetAddress getLocalAddress()
+	   {
+	      if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) SibTr.entry(this, tc, "getLocalAddress");
+	      InetAddress localAddress = ((InetSocketAddress) this.conn.getVirtualConnection().localAddress()).getAddress();
+	      if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) SibTr.exit(this, tc, "getLocalAddress", localAddress);
+	      return localAddress;
+	   }
+
+	   /**
+	    *
+	    * @see com.ibm.ws.sib.jfapchannel.framework.IOConnectionContext#getRemoteAddress()
+	    */
+	   public InetAddress getRemoteAddress()
+	   {
+	      if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) SibTr.entry(this, tc, "getRemoteAddress");
+	      InetAddress remoteAddress = ((InetSocketAddress) this.conn.getVirtualConnection().remoteAddress()).getAddress();
+	      if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) SibTr.exit(this, tc, "getRemoteAddress", remoteAddress);
+	      return remoteAddress;
+	   }
+
+	   /**
+	    *
+	    * @see com.ibm.ws.sib.jfapchannel.framework.IOConnectionContext#getLocalPort()
+	    */
+	   public int getLocalPort()
+	   {
+	      if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) SibTr.entry(this, tc, "getLocalPort");
+	      int localPort = ((InetSocketAddress) this.conn.getVirtualConnection().localAddress()).getPort();
+	      if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) SibTr.exit(this, tc, "getLocalPort", Integer.valueOf(localPort));
+	      return localPort;
+	   }
+
+	   /**
+	    *
+	    * @see com.ibm.ws.sib.jfapchannel.framework.IOConnectionContext#getRemotePort()
+	    */
+	   public int getRemotePort()
+	   {
+	      if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) SibTr.entry(this, tc, "getRemotePort");
+	      int remotePort = ((InetSocketAddress) this.conn.getVirtualConnection().remoteAddress()).getPort();
+	      if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) SibTr.exit(this, tc, "getRemotePort", Integer.valueOf(remotePort));
+	      return remotePort;
+	   }
+
+	   /**
+	    *
+	    * @see com.ibm.ws.sib.jfapchannel.framework.IOConnectionContext#getReadInterface()
+	    */
+	   public IOReadRequestContext getReadInterface()
```

Comment:
Odd that the method name refers to Interface but all the rest of the impl code uses context?